projects/observability/src/shared/dashboard/widgets/topology/metric/edge-metric-category.ts

```diff
@@ -0,0 +1,129 @@
+import { Color } from '@hypertrace/common';
+import { isEmpty } from 'lodash-es';
+
+export interface EdgeMetricCategory {
+  value: EdgeMetricCategoryValueType;
+  categoryClass?: string; // Can be used as selector class
+  color: string; // Color for edge, arrow and bubble
+}
+
+export type EdgeMetricCategoryValueType = PrimaryEdgeMetricCategoryValueType | SecondaryEdgeMetricCategoryValueType;
+
+export enum PrimaryEdgeMetricCategoryValueType {
+  LessThan20 = 'less-than-20',
+  From20To100 = 'from-20-to-100',
+  From100To500 = 'from-100-to-500',
+  From500To1000 = 'from-500-to-1000',
+  GreaterThanOrEqualTo1000 = 'greater-than-or-equal-to-1000',
+  NotSpecified = 'not-specified'
+}
+
+export enum SecondaryEdgeMetricCategoryValueType {
+  LessThan5 = 'less-than-5',
+  GreaterThanOrEqualTo5 = 'greater-than-or-equal-to-5',
+  NotSpecified = 'not-specified'
+}
+
+export const defaultPrimaryEdgeMetricCategories: EdgeMetricCategory[] = [
```

Comment:
Shouldn't these be coming from the model json. May be we define a `DefaultTopologyNodeMetricsModel` and a `DefaultTopologyEdgeMetricsModel` that can store these values. Ofcouse they should extend from `TopologyMetricsModel` defined above.

Reply:
I don't think we should define this, Because it just doesn't serve the purpose of having default categories.
I mean, imo default categories will act something that we don't need to put inside the metrics object to pass them.